cpp/test/neighbors/ann_ivf_flat.cuh

```diff
@@ -252,6 +254,124 @@ class AnnIVFFlatTest : public ::testing::TestWithParam<AnnIvfFlatInputs<IdxT>> {
     }
   }
 
+  void testIVFFlatCosine()
```

Comment:
Initially because the naive knn used in the tests didn't support cosine so I switched to brute-force to test.
I now added cosine to the naive knn so the regular test code is now used.